base/vcatobj/vcatobj.man
.ds Vn 1.12
.TH vcatobj 1Vi "24 April 1993" "Vista Version \*(Vn"
.SH NAME
vcatobj \- concatenate Vista data files
.SH SYNOPSIS
\fBvcatobj\fR [\fB-\fIoption\fR ...] [\fIinfile\fR ...] [\fB> \fIoutfile\fR]
.SH DESCRIPTION
\fBvcatobj\fP copies the contents of one or more input files to an output file.
All objects found in all input files (such as images, edge lists, and
comments) are copied, in order, to the single output file.
.SH "COMMAND LINE OPTIONS"
\fBvcatobj\fP accepts the following options:
.IP \fB-help\fP 15n
Prints a message describing options.
.IP "\fB-in\fP \fIinfile\fP ..."
Specifies one or Vista data files to be input.
.IP "\fB-out\fP \fIoutfile\fP"
Specifies where to write the output as a Vista data file.
.PP
Input files can be specified on the command line or allowed to default to
the standard input stream. The output file can be specified by the
\fB-out\fP option or allowed to default to the standard output stream.
.SH "SEE ALSO"
.BR vcatbands (1Vi),
.BR Vista (7Vi)
.SH NOTES
All input files are read before the output file is written. The program vcatobj was originally be known as vcat, created
by Art Pope.
.SH AUTHOR
Art Pope <pope@cs.ubc.ca>


debian/changelog
via (2.0.1svn191-0) stable; urgency=low

  *renamed via-bin to via-utils

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400 

via (2.0.0-0) stable; urgency=low

  *This version comes with the new lipsia-2.0 package
   and comprises some minor bug fixes
  *Fixed type error of VLong (is now long)

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400 

via (1.6.0-2) unstable; urgency=low

  * New internal release

 -- Frank Burkhardt <burk@cbs.mpg.de>  Mon, 20 Jul 2009 11:45:41 +0200

via (1.6.0-1) unstable; urgency=low

  * Initial release (Closes: #521328).
  * New Upstream Version. Renamed 'vcat' to 'vcatobj' to not conflict with
    atfs package in Debian.
  * Update Standards-version to 3.8.2.0 -- no changes necessary.
  * Update watchfile for recent change in upstream naming scheme.
  * Add versioned dependency on debhelper (>= 5) upon recommendation by
    lintian.

 -- Michael Hanke <michael.hanke@gmail.com>  Fri, 19 Jun 2009 13:50:52 +0200

via (1.5.2+svn3529-2) UNRELEASED; urgency=low

  * Add alternative build-dependency on atlas3-base-dev to be able to build
    the package on Debian etch.
  * FTP access to the upstream sources needs passive mode. Fixed debian/watch.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 14 Apr 2009 02:46:10 +0200

via (1.5.2+svn3529-1) UNRELEASED; urgency=low

  * New Upstream SVN snapshot that relocates the VIA lib headers to their new
    cannonical location.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 31 Mar 2009 22:08:01 +0200

via (1.5.2-1) UNRELEASED; urgency=low

  * Initial Debian packaging. This is a complete rewrite of the original
    Debian packaging effort done by the upstream developers.

 -- Michael Hanke <michael.hanke@gmail.com>  Thu, 26 Mar 2009 19:46:09 +0100


base/vcatobj/vcatobj.c
/****************************************************************
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 *
 * $Id: vcatobj.c 3177 2008-04-01 14:47:24Z karstenm $
 *
 *****************************************************************/

/*! \brief vcatobj - concatenates Vista data files

\par Description
concatenates Vista data files.

\par Usage

        <code>vcatobj</code>

        \param -in      input image(s)
        \param -out     output image

\par Examples
<br>
\par Known bugs
none.

\file vcatobj.c
\author Arthur Pope, UBC Laboratory for Computational Intelligence
*/


/*
 *
 *  The vcatobj program, which concatenates Vista data files.
 */

/*
 *  Copyright 1993, 1994 University of British Columbia
 *
 *  Permission to use, copy, modify, distribute, and sell this software and its
 *  documentation for any purpose is hereby granted without fee, provided that
 *  the above copyright notice appears in all copies and that both that
 *  copyright notice and this permission notice appear in supporting
 *  documentation. UBC makes no representations about the suitability of this
 *  software for any purpose. It is provided "as is" without express or
 *  implied warranty.
 *
 *  Author: Arthur Pope, UBC Laboratory for Computational Intelligence
 */

/* From the Vista library: */
#include "viaio/Vlib.h"
#include "viaio/file.h"
#include "viaio/mu.h"
#include "viaio/option.h"
#include "viaio/os.h"


/*
 *  Program entry point.   
 */

int main (int argc, char *argv[])
{
    static VArgVector in_files;
    static VString out_filename;
    static VBoolean in_found, out_found;
    static VOptionDescRec options[] = {
	{ "in", VStringRepn, 0, & in_files, & in_found, NULL,
	      "Input files" },
	{ "out", VStringRepn, 1, & out_filename, & out_found, NULL,
	      "Output file" }
    };
    FILE *f;
    VStringConst in_filename;
    VAttrList in_list, out_list = VCreateAttrList ();
    VAttrListPosn posn;
    int i, n;
    char prg[50];	
    sprintf(prg,"vcatobj V%s", getVersion());
    fprintf (stderr, "%s\n", prg);


    /* Parse command line arguments: */
    if (! VParseCommand (VNumber (options), options, & argc, argv) ||
        ! VIdentifyFiles (VNumber (options), options, "in", & argc, argv, 0) ||
	! VIdentifyFiles (VNumber (options), options, "out", & argc, argv, -1))
	goto Usage;
    if (argc > 1) {
	VReportBadArgs (argc, argv);
Usage:	VReportUsage (argv[0], VNumber (options), options, NULL);
	exit (EXIT_FAILURE);
    }

    /* For each input file: */
    for (i = 0; i < in_files.number; i++) {
	in_filename = ((VStringConst *) in_files.vector)[i];

	/* Read its contents: */
	if (strcmp (in_filename, "-") == 0)
	    f = stdin;
	else {
	    f = fopen (in_filename, "r");
	    if (! f)
		VError ("Failed to open input file %s", in_filename);
	}
	if (! (in_list = VReadFile (f, NULL)))
	    exit (EXIT_FAILURE);
	fclose (f);

	/* Append the input file's contents to the output list: */
	if (out_list->prev)
	    out_list->prev->next = in_list->next;
	else out_list->next = in_list->next;
	out_list->prev = in_list->prev;
	in_list->next = in_list->prev = NULL;
	VDestroyAttrList (in_list);
    }

    /* Count the number of attributes now in the output list: */
    for (n = 0, VFirstAttr (out_list, & posn); VAttrExists (& posn);
	 VNextAttr (& posn), n++) ;

    /* Open and write the output file: */
    if (strcmp (out_filename, "-") == 0)
	f = stdout;
    else {
	f = fopen (out_filename, "w");
	if (! f)
	    VError ("Failed to open output file %s", out_filename);
    }
    if (! VWriteFile (f, out_list))
	exit (EXIT_FAILURE);

    fprintf (stderr, "%s: Copied %d object%s.\n",
	     argv[0], n, n == 1 ? "" : "s");
    return EXIT_SUCCESS;
}
